### ISSUE TITLE
allow more data types in json responses

### ISSUE THREAD

**Issue by chpio** [rustless/rustless#71]
https://github.com/rustless/rustless/blob/0929e59d9c3c32f3024bf8e9de67d9b72ab82b49/src/framework/client.rs#L91-L96

some kind of:

``` rust
use serde_json::ser::to_string;
use serde::ser::Serialize;

fn json<T>(mut self, result: &T) -> ClientResult<'a>
  where
    T: Serialize,
{
  self.set_json_content_type();
  self.response.replace_body(Box::new(to_string(result).unwrap()));
  Ok(self)
}
```

**Comment by chpio** [comment]
this should also fix #64

**Comment by s-panferov** [comment]
@chpio yes, I'll try to implement this closer to the weekend
